lang/objects/object_builder.rs

```diff
@@ -36,14 +50,14 @@ impl ObjectBuilder {
     }
 
     /// Add attributes to object.
-    pub fn add_attributes(&mut self, attributes: ObjectAttributes) -> &mut Self {
+    pub fn add_attributes(&mut self, attributes: MetaData) -> &mut Self {
```

Comment:
this does not *add* attributes, instead it overwrites or *set*s the attributes.
```suggestion
    pub fn set_attributes(&mut self, attributes: MetaData) -> &mut Self {
```